src/ui/components/Library/RecordingOptionsDropdown.tsx

```diff
@@ -8,28 +8,47 @@ import MaterialIcon from "../shared/MaterialIcon";
 import hooks from "ui/hooks";
 import { RecordingId } from "@recordreplay/protocol";
 import { WorkspaceId } from "ui/state/app";
-import { Dropdown, DropdownDivider, DropdownItem } from "./LibraryDropdown";
+import { Dropdown, DropdownItem } from "./LibraryDropdown";
 import PortalDropdown from "../shared/PortalDropdown";
 import classNames from "classnames";
 import MoveRecordingMenu from "./MoveRecordingMenu";
 import { useConfirm } from "../shared/Confirm";
 import { isPublicDisabled } from "ui/utils/org";
+import { useGetNonPendingWorkspaces } from "ui/hooks/workspaces";
 
 type RecordingOptionsDropdownProps = PropsFromRedux & {
   recording: Recording;
 };
 
+function useGetShowDropdown(recording: Recording) {
+  const { userId, loading } = hooks.useGetUserId();
+  const { workspaces } = useGetNonPendingWorkspaces();
+
+  if (loading) {
+    return { showDropdown: false, loading };
+  }
+
+  const isOwner = userId == recording.user?.id;
+  // Node recordings don't have an owner since they're uploaded using the workspace's
+  // API key. We add this check here so the team is able to access the dropdown.
+  const sameTeam =
+    recording.workspace?.id && workspaces.find(w => w.id === recording.workspace?.id);
+  const showDropdown = isOwner || (!recording.user && sameTeam);
+
+  return { showDropdown, loading };
+}
+
 function RecordingOptionsDropdown({
   currentWorkspaceId,
   recording,
   setModal,
 }: RecordingOptionsDropdownProps) {
   const [isPrivate, setIsPrivate] = useState(recording.private);
   const [expanded, setExpanded] = useState(false);
+  const { showDropdown, loading: loading1 } = useGetShowDropdown(recording);
   const deleteRecording = hooks.useDeleteRecordingFromLibrary();
-  const { workspaces, loading } = hooks.useGetNonPendingWorkspaces();
+  const { workspaces, loading: loading2 } = hooks.useGetNonPendingWorkspaces();
```

Comment:
`loading1` can then become `loadingOwnershipInfo` or so, and `loading2` `loadingWorkspaces`.

I'd strongly recommend against using numbers, as they carry very little information and usually just make things more confusing.